src/model/MutableAreaDataSource.ts

```diff
@@ -301,7 +301,21 @@ export default class MutableAreaDataSource extends AreaDataSource {
         throw new Error('Area update error.  Reason: Updating leaf or boulder status of an area with subareas is not allowed.')
       }
 
-      if (areaName != null) area.set({ area_name: sanitizeStrict(areaName) })
+      if (areaName != null) {
+        const sanitizedName = sanitizeStrict(areaName)
+        area.set({ area_name: sanitizedName })
+
+        // change our pathTokens
+        const newPath = [...area.pathTokens]
+        newPath[newPath.length - 1] = sanitizedName
+        area.set({ pathTokens: newPath })
+
+        // iterate over the DB id's in the child list and update pathTokens
+        for (const childId of area.children) {
```

Comment:
You may be able to eliminate the for loop and multiple find() calls by either:

- Calling mongoose populate() function to turn the array of area IDs into array of area objects.
- Using the $in operator to get all child area objects in 1 call

Similar code
https://github.com/OpenBeta/openbeta-graphql/blob/develop/src/db/utils/jobs/TreeUpdater.ts#L47

This is just a comment, not a request to change.  If you want, you can do performance optimization in a follow up PR.

Reply:
I've pushed a small modification to `updatePathTokens()`, calling `populate()` to hydrate area.children[] to avoid calling find() in a for loop. I renamed Oregon state and saw an improvement from 5s down to 2s.